constructor.sh

```diff
@@ -107,6 +107,10 @@ pushd ${SCRIPT_DIR} >/dev/null
 . $1
 popd >/dev/null
 
+if [[ ! -z "$GIT_DEMOS_BRANCH_OPT" ]]; then
```

Comment:
If we pass in GIT_DEMOS_BRANCH rather than GIT_DEMOS_BRANCH_OPT we don't need this section of code at all, because line 107 above will override GIT_DEMOS_BRANCH automagically.

Reply:
(It also helps with the rebasing...) :-)